labelindex/label_inheritance_index.go

```diff
@@ -18,32 +18,62 @@ import (
 	log "github.com/Sirupsen/logrus"
 
 	"github.com/projectcalico/felix/dispatcher"
-	"github.com/projectcalico/felix/multidict"
+	"github.com/projectcalico/felix/set"
 	"github.com/projectcalico/libcalico-go/lib/backend/api"
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	"github.com/projectcalico/libcalico-go/lib/selector"
 )
 
+type itemData struct {
+	labels  map[string]string
+	parents []*parentData
+}
+
+// Get implements the Labels interface for itemData.  Combines the item's own labels with that
+// of its parent on the fly.
+func (itemData *itemData) Get(labelName string) (value string, present bool) {
+	if value, present = itemData.labels[labelName]; present {
+		return
+	}
+	for _, parent := range itemData.parents {
+		if value, present = parent.labels[labelName]; present {
+			return
+		}
+		for _, tag := range parent.tags {
+			if tag == labelName {
+				present = true
```

Comment:
I think we can rely on the go convention of returning the zero value with an error.